components/BarChartYears/index.tsx

```diff
@@ -218,3 +185,41 @@ function BarChartYears({ data }: BarChartProps) {
 }
 
 export default BarChartYears
+
+type GroupedByYear = Record<
+    string,
+    Array<{
+        attorney_type: string
+        year: number
+        case_count: number
+    }>
+>
+
+type TotalsByYear = Record<
+    string,
+    {
+        year: string
+        retained: number
+        appointed: number
+    }
+>
+
+const getTotals = (r: GroupedByYear): TotalsByYear => {
+    return mapWithIndex(r, (year, stats) => {
+        const groupedByAttorneyType = groupBy(stats)((s) => s.attorney_type)
+
+        const r1 = groupedByAttorneyType['Retained'][0].case_count
+        const r2 = groupedByAttorneyType['Retained'][1].case_count
+        const c1 = groupedByAttorneyType['Court Appointed'][0].case_count
+        const c2 = groupedByAttorneyType['Court Appointed'][1].case_count
+
+        const retained = (r1 > r2 ? r2 / r1 : r1 / r2) * 100
+        const appointed = (c1 > c2 ? c2 / c1 : c1 / c2) * 100
+
+        return {
+            year,
+            retained,
+            appointed,
+        }
+    })
+}
```

Comment:
This is great! so much faster at loading this graph and simplified the code, love it

Reply:
I think we could refine the query to be even closer to the final shape that we want... I wish my sql chops were better! (halp me ChatGPT!)